src/jllvm/vm/InteropHelpers.hpp

```diff
@@ -47,9 +47,9 @@ concept JavaConvertible = !std::is_void_v<T> && JavaCompatible<JavaConvertedType
 /// Helper function to call 'fnPtr', which is known to be a Java function, with the given 'args'.
 /// Does implicit conversion of 'args' their 'JavaCompatible' type.
 template <JavaCompatible Ret, JavaConvertible... Args>
-Ret invokeJava(llvm::JITEvaluatedSymbol fnPtr, Args... args)
+Ret invokeJava(void* fnPtr, Args... args)
```

Comment:
Is it not possible to make it more typesafe?

Reply:
The point of the function is to cast the `fnPtr` according to `Ret` and `Args` where the actual function pointer type being cast to is a function of `Args`. 
Making it type-safe by requiring a specific function pointer type would not work since the point of the function is to compute this function pointer type that `fnPtr` should be cast to (and then perform the cast + call).